linera-base/src/identifiers.rs

```diff
@@ -324,6 +324,33 @@ pub enum GenericApplicationId {
     User(ApplicationId),
 }
 
+impl fmt::Display for GenericApplicationId {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            GenericApplicationId::System => Display::fmt("System", f),
+            GenericApplicationId::User(application_id) => {
+                Display::fmt("User", f)?;
```

Comment:
Should we add a separator (colon?) after `System` and `User`? Looks weird if the word is not separated from the hex string at all.